CS483/CS483/Kartaclysm/Components/ComponentRacer.cpp

```diff
@@ -0,0 +1,64 @@
+//----------------------------------------------------------------------------
+// ComponentRacer
+// Author: Matthew White
+//
+// Component that handles racer logic
+//----------------------------------------------------------------------------
+
+#include "ComponentRacer.h"
+
+namespace Kartaclysm
+{
+	ComponentRacer::ComponentRacer(
+		HeatStroke::GameObject* p_pGameObject)
+		:
+		Component(p_pGameObject),
+		m_pDriver(nullptr),
+		m_pKart(nullptr),
+		m_iCurrentLap(0),
+		m_iCurrentPosition(1),
+		m_iCurrentTrackPiece(0)
+	{
+		std::function<void(const HeatStroke::Event*)>* lapCompleteDelegate = new std::function<void(const HeatStroke::Event*)>(std::bind(&ComponentRacer::FinishLap, this, std::placeholders::_1));
```

Comment:
It kills me every time I see "new" being called without a delete. Start race, quit race, start race, quit race, memory leak all day long. You should unregister and delete in the deconstructor.